lib/rules/no-implicit-globals.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * @fileoverview Rule to check for implicit global variables and functions.
+ * @author Joshua Peek
+ * @copyright 2015 Joshua Peek. All rights reserved.
+ * See LICENSE file in root directory for full license.
+*/
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function(context) {
+    return {
+        "Program": function() {
+            var scope = context.getScope();
+
+            scope.variables.forEach(function(variable) {
```

Comment:
@josh From the spec,

> The LexicalEnvironment and VariableEnvironment components of an execution context are always Lexical Environments. When an execution context is created its LexicalEnvironment and VariableEnvironment components initially have the same value.

Both are Lexical Environments. Some declarations add bindings to one environment (const and let) and others add bindings to another (var) *as well as* create properties on the global object when they are at the top level. So escope's "Scope" objects are simply representing each Lexical Environment, regardless of whether they are the LexicalEnvionment or VariableEnvironment components of the execution context.